src/Agda2Hs/Compile/Term.hs

```diff
@@ -68,7 +73,18 @@ ifThenElse _ es = compileElims es >>= \case
   -- fully applied
   b : t : f : es' -> return $ Hs.If () b t f `eApp` es'
   -- partially applied
-  _ -> genericError $ "if_then_else must be fully applied"
+  _ -> genericError $ "if_then_else_ must be fully applied"
+
+expTypeSig :: QName -> Elims -> C (Hs.Exp ())
+expTypeSig _ es = do
+  let args = fromMaybe __IMPOSSIBLE__ $ allApplyElims es
+  case args of
+    _ : typArg : expArg : args' -> do         -- the first one is the level; we throw that away
+      exp <- fromMaybe __IMPOSSIBLE__ <$> compileArg expArg -- this throws an error if it was Nothing
```

Comment:
You're better off just doing `compileTerm . unArg`, which would avoid the call to `__IMPOSSIBLE__`